Automattic/woocommerce-payments
# Avoid checking jetpack_tos_agreed flag in platform-checkout tracking

Fixes #3902 

#### Changes proposed in this Pull Request
`jetpack-tracking` package checks for the presence of jetpack_tos_agreed in two places. 
1. In `should_enable_tracking` method in `Tracking` class 
2. In `record_event` [method](https://github.com/Automattic/jetpack-tracking/blob/master/legacy/class-jetpack-tracks-client.php#L68) under the legacy implementation which `Tracking` relies on.

In order to bypass these checks, this PR proposes extending WCPay's `Platform_Checkout_Tracker` from `Jetpack_Tracks_Client` instead of `Tracking`. This gives us more control over the tracking conditions.

Additionally, `is_offline_mode` [check](https://github.com/Automattic/jetpack-tracking/blob/48c6b9c71017585ecb0171fc9034ad48adc71b0e/src/class-tracking.php#L236) (which checks for the presence of [Jetpack debug flags ](https://github.com/Automattic/jetpack-status/blob/c339904491b53319bff13aaa77bd74b39c011617/src/class-status.php#L39)to bail out of tracking) is also removed. 








#### Testing instructions






**Setup**

* In your `wp_options` table.
  * Ensure `jetpack_tos_agreed` is **not** set or set to 0.
  * `_wcpay_feature_platform_checkout` option is set to 1.
* Enable Platform Checkout option as a merchant from Payments -> Settings -> Express checkouts.

**Test tracking via PHP**
* Add a product to the cart
* Visit the checkout page
* After five minutes, check the Tracks Live View and see that a `woocommerceanalytics_order_checkout_start` event was recorded with your site's blog ID.

**Test Ajax tracking**
* Add a product to the cart
* Visit the classic checkout 
* Enter an email address of an existing platform-checkout user
* After five minutes, check the Tracks Live View and see that a `woocommerceanalytics_platform_checkout_otp_prompt_start` event was recorded with your site's blog ID.
-------------------


**Post merge**
